sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -1837,13 +1839,19 @@ class Analyzer(
                   }
                   AggregateExpression(agg, Complete, isDistinct, filter)
                 // This function is not an aggregate function, just return the resolved one.
-                case other =>
-                  if (isDistinct || filter.isDefined) {
-                    failAnalysis("DISTINCT or FILTER specified, " +
-                      s"but ${other.prettyName} is not an aggregate function")
-                  } else {
-                    other
+                case other if (isDistinct || filter.isDefined) =>
+                  failAnalysis("DISTINCT or FILTER specified, " +
+                    s"but ${other.prettyName} is not an aggregate function")
+                case e: String2TrimExpression if arguments.size == 2 =>
+                  if (trimWarningEnabled.get) {
+                    log.warn("Two-parameter TRIM/LTRIM/RTRIM function signatures are deprecated." +
+                      s" Use SQL syntax `${e.prettyName.toUpperCase(Locale.ROOT)}(trimStr" +
+                      s" FROM str)` instead of `${e.prettyName}`. Please refer SPARK-28093.")
+                    trimWarningEnabled.set(false)
```

Comment:
Is this a suitable place for the warning? How about putting this in `CheckAnalysis` (this is not an error issue though)?

Reply:
If we put it in `CheckAnalysis`, we can't distinguish `trim(a, b)` and `trim(a from b)`. They are both `String2TrimExpression`

Reply:
Hi, @maropu . Since `CheckAnalysis` received already resolved expressions, there is no way to distinguish `SQL syntax TRIM(trimStr FROM str)` and `function invocation TRIM(trimStr, str)`.

Reply:
Oops. I was late. Yes. @cloud-fan 's comment is correct.

Reply:
Ah, I see.